pages/lib/refinery/pages/finder.rb

```diff
@@ -51,7 +51,7 @@ def translations_conditions(original_conditions)
         translations_conditions = {}
         original_conditions.keys.each do |key|
           if translated_attributes.include? key.to_s
-            translations_conditions["#{Page.translation_class.table_name}.#{key}"] = original_conditions.delete(key)
+            translations_conditions["#{Page::Translation.table_name}.#{key}"] = original_conditions.delete(key)
```

Comment:
Do you even need to do this? Both Globalize and Mobility (with i18n scope) should seemlessly handle translated attributes as if they were normal attributes. What Refinery is doing here is basically what Mobility is doing [here](https://github.com/shioyama/mobility/blob/b42686f65fa3eb98ed1c9bded6ded8a1c816f2d0/lib/mobility/backends/active_record/table/query_methods.rb#L97), which it automatically does when you call `Post.i18n.where(title: "foo")` etc.

Reply:
;tldr; I don't think you should really need the `translations_conditions` with either Globalize or Mobility.

Reply:
You probably right, it's the first time i dig into the i18n logic of Refinery, there is probably legacy things we don't ever need now.

Reply:
@bricesanchez do you want to try removing it? I believe it was just a performance optimisation to get everything loaded in the single query.

Reply:
@parndt i had a lot of failling specs (108 fails) when i tried to removed this method: 

```
  108) Crudify xhr_paging performs ajax paging of index
       Failure/Error: stmt = SQLite3::Statement.new( self, sql )

       ActiveRecord::StatementInvalid:
         SQLite3::SQLException: no such column: refinery_pages.locale: SELECT  "refinery_pages".* FROM "refinery_pages" INNER JOIN "refinery_page_translations" "translations_refinery_pages" ON "translations_refinery_pages"."refinery_page_id" = "refinery_pages"."id" INNER JOIN "refinery_page_translations" ON "refinery_page_translations"."refinery_page_id" = "refinery_pages"."id" AND "refinery_page_translations"."locale" = 'en' WHERE "refinery_pages"."locale" = 'en' AND "refinery_pages"."parent_id" IS NULL AND "refinery_page_translations"."slug" = ? ORDER BY "refinery_pages"."id" ASC LIMIT ?
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:91:in `initialize'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:91:in `new'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:91:in `prepare'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
       # ./pages/lib/refinery/pages/finder.rb:122:in `parent_page'
       # ./pages/lib/refinery/pages/finder.rb:106:in `find'
       # ./pages/lib/refinery/pages/finder.rb:83:in `find'
       # ./pages/lib/refinery/pages/finder.rb:149:in `find'
       # ./pages/lib/refinery/pages/finder.rb:9:in `by_path_or_id'
       # ./pages/app/models/refinery/page.rb:80:in `find_by_path_or_id'
       # ./pages/app/controllers/refinery/pages_controller.rb:110:in `call'
       # ./pages/app/controllers/refinery/pages_controller.rb:75:in `find_page'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/request_store-1.4.1/lib/request_store/middleware.rb:19:in `call'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/2.5.0/webrick/httpserver.rb:140:in `service'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/2.5.0/webrick/httpserver.rb:96:in `run'
       # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/2.5.0/webrick/server.rb:307:in `block in start_thread'
       # ------------------
       # --- Caused by: ---
       # SQLite3::SQLException:
       #   no such column: refinery_pages.locale
       #   /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:91:in `initialize'
```

I had seen that i will need this method so i had keep this logic.

Reply:
Ok, I figured out what's going on. The problem is that the original `conditions` hash has a `locale` key with an array of locales as its value. `translations_conditions` passes this on to the translations table, which uses it in an `IN` clause, but Mobility doesn't handle this one so you get an exception about a missing `locale` column on the parent model table.

This whole method can still be removed though, if you just update `with_mobility` to look like this:

```ruby
def with_mobility
  mobility_conditions = conditions.dup
  locales = mobility_conditions.delete(:locale)

  # A join implies readonly which we don't really want.
  Page.i18n.where(mobility_conditions).where(Page::Translation.table_name.to_sym => { locale: locales }).readonly(false)
end
def with_mobility
  mobility_conditions = conditions.dup
  locales = mobility_conditions.delete(:locale)

  # A join implies readonly which we don't really want.
  Page.i18n.where(mobility_conditions).
    where(Page::Translation.table_name.to_sym => { locale: locales }).readonly(false)
end
```

Reply:
Note that you don't need `joins(:translations)` here, Mobility will do that for you in the `i18n` scope.

Reply:
Actually wait, that doesn't quite work...

Reply:
Ok, long story short, this requires shioyama/mobility#51 to really work. The problem is that Mobility joins *ON* the current locale, and this clobbers this refinery query so you can't query on more than one locale. So for now, I'd say just leave this, and once shioyama/mobility#51 is done it should be refactorable.

Reply:
btw, good test coverage! Only 5 tests ended up failing, but they highlighted the problem here.

Reply:
I'm glad! 🥇